### PR TITLE
Added dropdown for current and battery meter selection to configuration tab

### PR DESCRIPTION
**Pull request by basdelfos**
Added dropdown for current and battery meter selection in configurator tab.

image

### CODE REVIEW

File: tabs/configuration.js [hunk 1]
```diff
@@ -96,17 +96,35 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
...
             MSP.send_message(MSPCodes.MSP_NAME, false, false, next_callback);
         } else {
             next_callback();
         }
     }
 
-    
+    function load_battery() {
+        var next_callback = load_current;
+        if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "3.1.0")) {
```
Comment (mikeller) [inline tabs/configuration.js]: The configurator now does a firmware check on connect, and disconnects if non - Betaflight is detected. Hence the `CONFIG.flightControllerIdentifier == "BTFL` check is not needed any more. It just hasn't been removed from all the places.

File: tabs/configuration.js [hunk 2]
```diff
@@ -410,28 +428,61 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
...
+        batteryMeterType_e.val(MISC.batterymetertype);
+
         $('input[name="mincellvoltage"]').val(MISC.vbatmincellvoltage);
         $('input[name="maxcellvoltage"]').val(MISC.vbatmaxcellvoltage);
         $('input[name="warningcellvoltage"]').val(MISC.vbatwarningcellvoltage);
         $('input[name="voltagescale"]').val(MISC.vbatscale);
 
         // fill current
+        var currentMeterTypes = [
+            'None', //Do not show in GUI
```
Comment (mikeller) [inline tabs/configuration.js]: `None` should be shown, as it is a valid setting.

File: tabs/configuration.js [hunk 3]
```diff
@@ -410,28 +428,61 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
...
         // fill throttle
         $('input[name="midrc"]').val(MISC.midrc);
         $('input[name="minthrottle"]').val(MISC.minthrottle);
         $('input[name="maxthrottle"]').val(MISC.maxthrottle);
         $('input[name="mincommand"]').val(MISC.mincommand);
-        
+
         // fill battery
+        var batteryMeterTypes = [
+            'None', //Do not show in GUI
```
Comment (mikeller) [inline tabs/configuration.js]: `None` should be removed altogether, see https://github.com/betaflight/betaflight/pull/1646.

File: js/msp/MSPHelper.js [hunk 4]
```diff
@@ -203,6 +203,21 @@ MspHelper.prototype.process_data = function(dataHandler) {
             MISC.vbatmaxcellvoltage = data.readU8() / 10; // 10-50
             MISC.vbatwarningcellvoltage = data.readU8() / 10; // 10-50
             break;
+        case MSPCodes.MSP_VOLTAGE_METER_CONFIG:
+            MISC.vbatscale = data.readU8(); // 10-200
+            MISC.vbatmincellvoltage = data.readU8() / 10; // 10-50
+            MISC.vbatmaxcellvoltage = data.readU8() / 10; // 10-50
+            MISC.vbatwarningcellvoltage = data.readU8() / 10; // 10-50
+            if (semver.gte(CONFIG.apiVersion, "1.23.0")) {
+                MISC.batteryMeterType = data.readU8();
```
Comment (mikeller) [inline js/msp/MSPHelper.js]: `MISC.batterymetertype` is lower case everywhere else, causing this to fail.

File: js/msp/MSPHelper.js [hunk 5]
```diff
@@ -1006,7 +1026,21 @@ MspHelper.prototype.crunch = function(code) {
...
                 .push8(Math.round(MISC.vbatwarningcellvoltage * 10));
             break;
-
+        case MSPCodes.MSP_SET_VOLTAGE_METER_CONFIG:
+            buffer.push8(MISC.vbatscale)
+                .push8(Math.round(MISC.vbatmincellvoltage * 10))
+                .push8(Math.round(MISC.vbatmaxcellvoltage * 10))
+                .push8(Math.round(MISC.vbatwarningcellvoltage * 10));
+                if (semver.gte(CONFIG.apiVersion, "1.23.0")) {
+                    buffer.push8(MISC.batteryMeterType);
```
Comment (mikeller) [inline js/msp/MSPHelper.js]: `MISC.batterymetertype` is lower case everywhere else, causing this to fail.

**Comment by basdelfos** [comment]
Fixed requested changes and made state of battery and current controls dependent from GUI selections

**Comment by mikeller** [comment]
Looking good, thanks for the changes, will give it another go tonight.

**Pull request merged**